webapp/app/js/compare.js

```diff
@@ -0,0 +1,448 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+
+// -------------------------------------------------------------------------
+// Utility Functions
+// -------------------------------------------------------------------------
+
+
+/**
+ * Compute the standard deviation for an array of values.
+ * 
+ * @param values
+ *        An array of numbers.
+ * @param avg
+ *        Average of the values.
+ * @return a number (the standard deviation)
+ */
+function stddev(values, avg) {
+  if (values.length <= 1) {
+    return 0;
+  }
+
+  return Math.sqrt(
+    values.map(function (v) { return Math.pow(v - avg, 2); })
+          .reduce(function (a, b) { return a + b; }) / (values.length - 1));
+}
+
+
+// -------------------------------------------------------------------------
+// End Utility Functions
+// -------------------------------------------------------------------------
+
+
+var compare = angular.module("compare", ['ui.router', 'ui.bootstrap', 'treeherder']);
+
+compare.factory('getSeriesSummary', [ function() {
+  return function(signature, signatureProps, optionCollectionMap, pgo, e10s) {
+    var platform = signatureProps.machine_platform + " " +
+      signatureProps.machine_architecture;
+    var extra = "";
+    if (signatureProps.job_group_symbol === "T-e10s") {
+      extra = " e10s";
+    }
+    var testName = signatureProps.test;
+    var subtestSignatures;
+    if (testName === undefined) {
+      testName = "summary";
+      subtestSignatures = signatureProps.subtest_signatures;
+    }
+    var name = signatureProps.suite + " " + testName +
+      " " + optionCollectionMap[signatureProps.option_collection_hash] + extra;
+
+    //Only keep summary signatures, filter in/out e10s and pgo
+    if (name.indexOf('summary') <= 0) {
+        return null;
+    }
+    if (e10s && (name.indexOf('e10s') <= 0)) {
+        return null;
+    } else if (!e10s && (name.indexOf('e10s') > 0)) {
+        return null;
+    }
+
+    //TODO: pgo is linux/windows only- what about osx and android
+    if (pgo && (name.indexOf('pgo') <= 1)) {
+        return null;
+    } else if (!pgo && (name.indexOf('pgo') > 0)) {
+        return null;
+    }
+
+    return { name: name, signature: signature, platform: platform,
+             subtestSignatures: subtestSignatures };
+  };
+}]);
+
+compare.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$rootScope', '$location',
+                              '$modal', 'thServiceDomain', '$http', '$q', '$timeout', 'getSeriesSummary',
+  function CompareCtrl($state, $stateParams, $scope, $rootScope, $location, $modal,
+                    thServiceDomain, $http, $q, $timeout, getSeriesSummary) {
+
+    function displayComparision() {
+      //TODO: why do we need so much history?
```

Comment:
I assume we need the history so we can dig out older results? We could probably do better by seeing how old each revision is and only fetching enough data to cover them both.